codex/codex.nim

```diff
@@ -221,9 +224,18 @@ proc new*(T: type CodexServer, config: CodexConf, privateKey: CodexPrivateKey):
       .expect("Should start rest server!")
 
   switch.mount(network)
-  T(
+
+  return CodexServer(
     config: config,
     codexNode: codexNode,
     restServer: restServer,
     repoStore: repoStore,
-    maintenance: maintenance)
+    maintenance: maintenance
+  )
+
+proc new*(
+    _: typedesc[CodexServer],
```

Comment:
I thought `typedesc[CodexServer]` was outdated in favour of `type CodexServer`?

Reply:
The `typedesc[Foo]` form is more common from what I've seen in most Nim libraries outside Status ones. Plus I believe that `type Foo` actually requires running `type` as a static function which returns `typedesc[Foo]`. I'll ask around about it.

Reply:
From what I can tell the recommendation appears to be `typedesc[T]`. It looks like the `type Foo` form is only used in a few status libs. I'd guess just because it's a slight bit easier, but it incurs a slight overhead in the compiler and overloads `type` more. 

Also, I searched through a dozen or so Nim libraries and the dominant usage was `newFoo`. I suspect because it avoids the generic overload and works better with nimsuggest type completetion. 

For now I backed out those changes though.

Reply:
I believe we are following the object construction recommendations in [the style guide](https://status-im.github.io/nim-style-guide/language.objconstr.html). If there are some improvements to be made to the style guide, it might be best to raise a PR in the style guide and have a discussion there. If those changes get merged, then we can go through the process of updating nim-codex.

Reply:
Thanks, that's a good idea.